org.eclipse.ice.commands/src/main/java/org/eclipse/ice/commands/LocalMoveFileCommand.java

```diff
@@ -68,45 +68,62 @@ public CommandStatus execute() {
 	@Override
 	protected CommandStatus run() {
 
-		// Check to see if the file paths are the same and if the filename has the same extension
-		// for determining whether or not it should be a move or change file name
+		// Check to see if the file paths are the same and if the filename has the same
+		// extension for determining whether or not it should be a move or change file
+		// name
 		boolean sameFileExtension = checkFileExtension();
 
 		// Split the destination by path delimiter to get the desired filename
 		String[] destinationDirs = destination.toString().split("/");
-		
+
 		// If the directory paths are the same and the files have the same file
 		// extension
 		// type, then we are just changing a file name
 		if (sameFileExtension) {
 			try {
+				status = CommandStatus.RUNNING;
 				// destinationDirs will have the desired file name in the length-1 entry
 				Files.move(source, source.resolveSibling(destinationDirs[destinationDirs.length - 1]),
 						REPLACE_EXISTING);
 			} catch (IOException e) {
 				e.printStackTrace();
+				status = CommandStatus.FAILED;
+				return status;
 			}
 		}
 		// All other cases the file is moving directory
 		else {
 			try {
-				Files.move(source, destination.resolve(source.getFileName()));
+				status = CommandStatus.RUNNING;
+				// Try to first move it as a new file name
+				Files.move(source, destination);
 			} catch (IOException e) {
-				e.printStackTrace();
+				try {
+					// If that catches, then try to move it into a different directory
+					// with the same name
+					Files.move(source, destination.resolve(source.getFileName()));
+				} catch (IOException e1) {
+					// If that catches, then it really failed and return as such
+					e.printStackTrace();
+					e1.printStackTrace();
```

Comment:
Remove these or log the exception.

Reply:
See above comment, went through and made sure all exceptions were logged.